components/ruby/lib/chef-licensing/restful_client/base.rb

```diff
@@ -75,43 +75,66 @@ def list_licenses(params = {})
 
       # a common method to handle the get API calls
       def invoke_get_api(endpoint, params = {})
-        handle_get_connection do |connection|
-          connection.get(endpoint, params).body
-        end
+        response = invoke_api(ChefLicensing::Config.license_server_url.split(","), endpoint, :get, nil, params)
+        response.body
       end
 
       # a common method to handle the post API calls
       def invoke_post_api(endpoint, payload, headers = {})
-        handle_post_connection do |connection|
-          response = connection.post(endpoint) do |request|
-            request.body = payload.to_json
-            request.headers = headers
+        response = invoke_api(ChefLicensing::Config.license_server_url.split(","), endpoint, :post, payload, nil, headers)
+        raise RestfulClientError, format_error_from(response) unless response.success?
+
+        response.body
+      end
+
+      def invoke_api(urls, endpoint, http_method, payload = nil, params = {}, headers = {})
+        handle_connection = http_method == :get ? method(:handle_get_connection) : method(:handle_post_connection)
+        response = nil
+        # Note: Current limit is set to 5 attempts for trying to connect to the server
```

Comment:
This comment sounds like there will be a minimum of 5 attempts to connect to the server(each). If I am not wrong the limit is on multiple server url that User provides right?